driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/VoidReadOperationThenCursorReadOperation.java

```diff
@@ -46,12 +49,17 @@ public String getCommandName() {
     }
 
     @Override
-    public void executeAsync(final AsyncReadBinding binding, final SingleResultCallback<AsyncBatchCursor<T>> callback) {
-        readOperation.executeAsync(binding, (result, t) -> {
+    public BatchCursor<T> execute(final ReadBinding binding, final OperationContext operationContext) {
```

Comment:
No need to explicitly add this as it should be in the ReadOperationCursorAsyncOnly - lets update the default method there instead.